Add in Val's text for connectivity matrices (#173)
PennLINC/qsirecon: docs/builtin_workflows.rst

\* Not recommended

.. _connectivity_matrices:

*********************
Connectivity matrices

> *********************

Reviewer: I just realized where you put this. Should this go in `outputs`?

Author: It's kind of outside of individual workflows (they all make these matfiles), but is slightly different for different workflows (the variables are named differently for dsi studio and mrtrix3)

.. code-block::

   qsirecon/
      derivatives/
         qsirecon-<suffix>/
            sub-<label>/
               dwi/
                  sub-<label>_space-T1w_seg-<label>_dseg.nii.gz
                  sub-<label>_space-T1w_seg-<label>_dseg.mif.gz
                  sub-<label>_space-T1w_seg-<label>_dseg.json
                  sub-<label>_space-T1w_seg-<label>_dseg.tsv

.. _custom_atlases:

Using custom atlases
--------------------

It's possible to use your own atlases provided you organize the atlases into BIDS-Atlas datasets.
Users can control which atlases are used with the ``--atlases`` and ``--datasets`` parameters.

The nifti images should be registered to the
`MNI152NLin2009cAsym <https://github.com/PennLINC/qsirecon/blob/main/qsirecon/data/mni_1mm_t1w_lps.nii.gz>`_
included in ``qsirecon``.
It is essential that your images are in the LPS+ orientation and have the sform zeroed-out in the header.
**Be sure to check for alignment and orientation** in your outputs.

Connectivity Measures
=====================

Connectivity measures are bundled together in binary ``.mat`` files,
rather than as atlas- and measure-specific tabular files.

.. warning::

>    We ultimately plan to organize the connectivity matrices accoring to the BIDS-Connectivity BEP,

Reviewer: ```suggestion
   We ultimately plan to organize the connectivity matrices according to the BIDS-Connectivity BEP,
```

Just noticed my own typo

\* Not recommended

.. _connectivity_matrices:

> #####################
> Connectivity matrices
> #####################

Reviewer: ```suggestion
*********************
Connectivity matrices
*********************
```

The section labeling needs to be adjusted. Here's the convention we have in conf.py.

````
# with overline, for parts
* with overline, for chapters
= for sections
- for subsections
^ for subsubsections
" for paragraphs
```

Instead of offering a bewildering number of options for constructing connectivity matrices,
``qsirecon`` will construct as many connectivity matrices as it can given the reconstruction
methods.
It is **highly** recommended that you pick a weighting scheme before you run
these pipelines and only look at those numbers.
If you look at more than one weighting method be sure to adjust your statistics for the
additional comparisons.

To skip this step in your workflow, you can modify an existing recon pipeline by removing the
``action: connectivity`` section from the yaml file.

.. _connectivity_atlases:

> *******
> Atlases
> *******

Reviewer: ```suggestion
Atlases
=======
```

QSIRecon, XCP-D, and ASLPrep, to produce synchronized outputs across modalities.
For more information about the 4S atlas, please see https://github.com/PennLINC/AtlasPack.

Atlases are written out to the ``atlases`` subfolder, following
`BEP038 <https://docs.google.com/document/d/1RxW4cARr3-EiBEcXjLpSIVidvnUSHE7yJCUY91i5TfM/edit?usp=sharing>`_.

.. code-block::

   qsirecon/
      atlases/
         dataset_description.json
         atlas-<label>/
            atlas-<label>_space-<label>_res-<label>_dseg.nii.gz
            atlas-<label>_space-<label>_res-<label>_dseg.json
            atlas-<label>_dseg.tsv

Additionally, each atlas is warped to the subject's anatomical space and written out in the
associated reconstruction workflows dataset.

.. code-block::

   qsirecon/
      derivatives/
         qsirecon-<suffix>/
            sub-<label>/
               dwi/
                  sub-<label>_space-T1w_seg-<label>_dseg.nii.gz
                  sub-<label>_space-T1w_seg-<label>_dseg.mif.gz
                  sub-<label>_space-T1w_seg-<label>_dseg.json
                  sub-<label>_space-T1w_seg-<label>_dseg.tsv

.. _custom_atlases:

> Using custom atlases
> ^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Using custom atlases
--------------------
```

It's possible to use your own atlases provided you organize the atlases into BIDS-Atlas datasets.
Users can control which atlases are used with the ``--atlases`` and ``--datasets`` parameters.

The nifti images should be registered to the
`MNI152NLin2009cAsym <https://github.com/PennLINC/qsirecon/blob/main/qsirecon/data/mni_1mm_t1w_lps.nii.gz>`_
included in ``qsirecon``.
It is essential that your images are in the LPS+ orientation and have the sform zeroed-out in the header.
**Be sure to check for alignment and orientation** in your outputs.

> *********************
> Connectivity Measures
> *********************

Reviewer: ```suggestion
Connectivity Measures
=====================
```

   wherein each measure from each atlas is stored in a separate file.

   Therefore, this organization will change in the future.

.. code-block::

   qsirecon/
      derivatives/
         qsirecon-<suffix>/
            sub-<label>/[ses-<label>/]
               dwi/
                  <source_entities>_connectivity.mat

The ``.mat`` file contains a dictionary with all of the connectivity measures specified
by the recon spec for all of the different atlases specified by the user.

For example, in the case where a user has selected a single atlas (``<atlas>``) and
the recon spec specifies a single connectivity measure (``<measure>``),
the ``.mat`` file will contain the following keys:

.. code-block::

   command                                # The command that was run
   atlas_<atlas>_region_ids               # The region ids for the atlas (1 x n_parcels array)
   atlas_<atlas>_region_labels            # The region labels for the atlas (1 x n_parcels array)
   atlas_<atlas>_<measure>_connectivity   # The connectivity matrix for the atlas and measure (n_parcels x n_parcels array)

MRtrix3 Connectivity Measures

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
-----------------------------
```

MRtrix3 connectivity workflows produce 4 structural connectome outputs for each atlas.
The 4 connectivity matrix outputs are

>    * *radiusN.count.connectivity*: raw streamline count based matrix
>    * *sift.radiusN.count.connectivity*: sift-weighted streamline count based matrix
>    * *radiusN.meanlength.connectivity*: a matrix containing mean length of raw streamlines
>    * *sift.radiusN.meanlength.connectivity*: a matrix containing mean length of sifted output

Reviewer: ```suggestion
   * *atlas_<atlas>_radius<N>.count.connectivity*: raw streamline count based matrix
   * *atlas_<atlas>_sift.radius<N>.count.connectivity*: sift-weighted streamline count based matrix
   * *atlas_<atlas>_radius<N>.meanlength.connectivity*: a matrix containing mean length of raw streamlines
   * *atlas_<atlas>_sift.radius<N>.meanlength.connectivity*: a matrix containing mean length of sifted output
```

I think that's what this should be, but I could be wrong.

Author: I think you're right

The number ``N`` in ``radiusN`` indicates how many mm the algorithm would search up from a
given streamline's endpoint for a cortical region. E.g., a radius of 2 indicates that
if a streamline ended before hitting gray matter, the search for a cortical
termination region could be up to 2mm from the endpoint.

DSI Studio Connectivity Measures

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
--------------------------------
```
